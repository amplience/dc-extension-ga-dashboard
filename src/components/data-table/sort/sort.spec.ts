import { render, fireEvent, screen } from '@testing-library/svelte';
import Sort from './sort.svelte';

describe('Sort', () => {
  it('should render the DataTable Sort component', () => {
    const { container } = render(Sort, { direction: 'asc' });

    expect(container.firstChild).toMatchSnapshot();
  });

  it('should call onClick function when click area is clicked', async () => {
    const onClick = jest.fn();
    render(Sort, {
      direction: 'asc',
      property: 'test-property',
      onClick,
    });

    const element = screen.getByTestId('data-table-sort-click-area');

    await fireEvent.click(element);

    expect(onClick).toHaveBeenCalledWith('test-property');
  });
});
