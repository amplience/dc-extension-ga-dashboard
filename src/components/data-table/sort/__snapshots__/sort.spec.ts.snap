// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sort should render the DataTable Sort component 1`] = `
<div>
  <div
    class="data-table-sort asc svelte-1xz2vfo"
  />
   
  <div
    class="data-table-sort-click-area svelte-1xz2vfo"
    data-testid="data-table-sort-click-area"
  />
</div>
`;
